torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -658,12 +658,14 @@ class PrioritizeLoad : public IRMutator {
 };
 
 std::string CudaCodeGen::GetUniqueFuncName(const std::string& func_prefix) {
-  // We are using a global counter here to make sure difference instances
-  // within CudaCodeGen have different names.
-  static int64_t counter = 0;
-  ++counter;
-  int64_t value = counter;
-  return func_prefix + "_" + std::to_string(value);
+  int64_t counter = 0;
```

Comment:
I am fine with this change. But if the number of kernels is large, this could be quadratic. Can we move the counter to the class object scope as well, similar to taken_func_names? Thanks!

Reply:
This will only be quadratic on number of kernels with the same body (executing the same operations), since the name is now reflecting operations we're doing in the body, so the collisions would actually be quite rare I think. Do you think we could really hit a scenario where we have a big number of identical kernels in the same program (note that in the current pipeline we usually have just one kernel in the program)?

Reply:
I agree with you here that its unlikely to be a factor, but the fix in this case is pretty simple (change the set to a map to int64) it might be worth doing anyway.